lib/hooks/use-form-validation.ts

```diff
@@ -0,0 +1,123 @@
+import { ValidationResult } from "@/lib/utils/validation";
+import { useCallback, useState } from "react";
+
+interface UseFormValidationOptions<T> {
+  initialValues: T;
+  validate: (values: T) => ValidationResult;
+  onSubmit: (values: T) => Promise<void> | void;
+}
+
+export function useFormValidation<T extends Record<string, any>>({
```

Comment:
We are already using [React Hook Form](https://react-hook-form.com/) and [Zod](https://zod.dev/) for form building.